compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1204,69 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        actual: Ty<'tcx>,
+        source: SelfSource<'tcx>,
+        span: Span,
+        item_name: Ident,
+        ty_str: &str,
+    ) -> bool {
+        if let SelfSource::MethodCall(expr) = source {
+            let mut search_limit = 5;
+            for (_, parent) in tcx.hir().parent_iter(expr.hir_id) {
+                search_limit -= 1;
+                if search_limit == 0 {
+                    break;
+                }
```

Comment:
is this intended to only do 4 loops instead of 5?
```suggestion
                if search_limit == 0 {
                    break;
                } else {
                    search_limit -= 5;
                }
```
alternatively, `for (_, parent) in tcx.hir().parent_iter(expr.hir_id).take(5)`

Reply:
Nice!

Reply:
> is this intended to only do 4 loops instead of 5?
not intended, 5 is also good I think.